Core/src/main/java/com/plotsquared/core/generator/HybridGen.java

```diff
@@ -217,34 +222,170 @@ public void generateChunk(@NonNull ScopedQueueCoordinator result, @NonNull PlotA
                         }
                         if (!hybridPlotWorld.ROAD_SCHEMATIC_ENABLED) {
                             if (hybridPlotWorld.PLACE_TOP_BLOCK) {
-                                result.setBlock(x, hybridPlotWorld.WALL_HEIGHT + 1, z,
-                                        hybridPlotWorld.WALL_BLOCK.toPattern()
-                                );
+                                result.setBlock(x, hybridPlotWorld.WALL_HEIGHT + 1, z, hybridPlotWorld.WALL_BLOCK.toPattern());
                             }
                         } else {
-                            placeSchem(hybridPlotWorld, result, relativeX[x], relativeZ[z], x, z,
-                                    true
-                            );
+                            placeSchem(hybridPlotWorld, result, relativeX[x], relativeZ[z], x, z, true, false);
                         }
                     } else {
                         // plot
                         for (int y = startY; y < hybridPlotWorld.PLOT_HEIGHT; y++) {
                             result.setBlock(x, y, z, hybridPlotWorld.MAIN_BLOCK.toPattern());
                         }
-                        result.setBlock(x, hybridPlotWorld.PLOT_HEIGHT, z,
-                                hybridPlotWorld.TOP_BLOCK.toPattern()
-                        );
+                        result.setBlock(x, hybridPlotWorld.PLOT_HEIGHT, z, hybridPlotWorld.TOP_BLOCK.toPattern());
                         if (hybridPlotWorld.PLOT_SCHEMATIC) {
-                            placeSchem(hybridPlotWorld, result, relativeX[x], relativeZ[z], x, z,
-                                    false
-                            );
+                            placeSchem(hybridPlotWorld, result, relativeX[x], relativeZ[z], x, z, false, false);
                         }
                     }
                 }
             }
         }
     }
 
+    @Override
+    public boolean populateChunk(final ScopedQueueCoordinator result, final PlotArea settings) {
+        HybridPlotWorld hybridPlotWorld = (HybridPlotWorld) settings;
+        if (!hybridPlotWorld.populationNeeded()) {
+            return false;
+        }
+        // Coords
+        Location min = result.getMin();
```

Comment:
Any objections about refactoring the new introduced stuff into private methods to reuse them if needed? I feel like we end up copy/pasting more than needed here lol

Reply:
I thought about it but imo it actually reduces the readability of the code given the overall differences, or requirements for multiple return values